app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,28 @@
+# require 'pry'
+class SessionsController < ApplicationController
+
+  def new
+    @user
+  end
+
+  def create
+    auth_hash = request.env['omniauth.auth']
+    if auth_hash[:uid]
+      @user = User.find_or_create_from_omniauth(auth_hash)
+      if @user
+        session[:user_id] = @user.id
+        redirect_to root_path
+      else
+        redirect_to root_path, notice: "Failed to save the user"
+      end
+    else
+      redirect_to root_path, notice: "Failed to authenticate"
+    end
+  end
+
+  def destroy
+    session.delete :user_id
+    redirect_to root_path
+  end
```

Comment:
Real talk: I just had to look through old projects because I haven't written Ruby in a couple months and wasn't positive what I used to do haha. I just knew it wasn't that. 

I've always just set it to nil: `session[:user_id] = nil` but I think your way looks nicer.